lib/shimmer/driver.rb

```diff
@@ -0,0 +1,195 @@
+require "capybara"
+require "chrome_remote"
+require "shimmer/browser"
+
+module Capybara
+  module Shimmer
+    class Driver < Capybara::Driver::Base
+      attr_reader :browser
+
+      # rubocop:disable Lint/UnusedMethodArgument
+      def initialize(app, options = {})
+        @app     = app
+        @options = options.dup
+        @browser = Capybara::Shimmer::Browser.new(@options).start
+      end
+
+      # def enable_logging
+      # end
+      #
+      # def allow_url(url)
+      # end
+      #
+      # def block_url(url)
+      # end
+      #
+      # def block_unknown_urls
+      # end
+      #
+      # def allow_unknown_urls
+      # end
+
+      def current_url
+        raise NotImplementedError
+      end
+
+      def visit(path)
+        browser.send_cmd "Page.navigate", url: path
+        # browser.wait_for "Page.lifecycleEvent", match: {"name" => "firstMeaningfulPaint"}
+
+        browser.wait_for_with_either_match("Page.lifecycleEvent",
```

Comment:
This was a major PITA. I tried my best to search for the right event callback. The thing is that not all the callbacks are guaranteed to fire, depending on whether the resource has been cached (so you can't depend on a Network HTTP lifecycle event). Some pages doesn't always seem to fire `load` either (!), so we fallback to `networkIdle` which means "I loaded everything and I'm waiting 🙄", which seems to reliably be a last resort.

I've also forked chrome_remote, and `wait_for*` should also take a timeout which we can implement in another story.